contrib/internal/httptrace/config_test.go

```diff
@@ -75,6 +76,87 @@ func TestConfig(t *testing.T) {
 	}
 }
 
+func TestIPCollection(t *testing.T) {
+	for _, tc := range []struct {
+		name      string
+		env       map[string]string
+		collectIP bool
+	}{
+		{
+			name:      "default-env",
+			collectIP: false,
+		},
+		{
+			name:      "disabled-collection",
+			env:       map[string]string{envClientIPHeaderDisabled: "true"},
+			collectIP: false,
+		},
+		{
+			name:      "enabled-collection",
+			env:       map[string]string{envClientIPHeaderDisabled: "false"},
+			collectIP: true,
+		},
+	} {
+		for k, v := range tc.env {
+			t.Setenv(k, v)
```

Comment:
I think the scope of this SetEnv is wrong here and should be in the t.Run closure below instead to scope it to the actual subtest

Reply:
cf998e7135e8fd38204eb73c227c27db2afaf12e